### PR TITLE
Highlight feature flags for debugging help in App Manager

### PR DESCRIPTION
**Pull request by biyeun**
- tags are only visible if the flag `Highlight / Mark Feature Flags in UI` is enabled
- not meant to look pretty, just for debugging between v2 and v1

screen shot 2017-03-10 at 3 32 41 pm

@orangejenny 
cc @benrudolph

### CODE REVIEW

File: corehq/apps/hqwebapp/templatetags/hq_shared_tags.py [hunk 1]
```diff
@@ -219,6 +225,26 @@ def toggle_enabled(request, toggle_or_toggle_name):
     import corehq.toggles
     return _toggle_enabled(corehq.toggles, request, toggle_or_toggle_name)
 
+@register.filter
```
Comment (dimagimon) [inline corehq/apps/hqwebapp/templatetags/hq_shared_tags.py]: E302 expected 2 blank lines, found 1

File: corehq/apps/hqwebapp/templatetags/hq_shared_tags.py [hunk 2]
```diff
@@ -219,6 +225,26 @@ def toggle_enabled(request, toggle_or_toggle_name):
...
+    and if not enabled, show where the UI would be.
+    Useful for trying to find out if you have all the flags enabled in a
+    particular location or whether a feature on prod is part of a particular
+    flag. """
+    if not toggles.SHOW_DEV_TOGGLE_INFO.enabled_for_request(request):
+        return ""
+    flag = _get_toggle(toggles, toggle_or_toggle_name)
+    tag = flag.tag
+    is_enabled = flag.enabled_for_request(request)
+    return mark_safe("""<div class="label label-{css_class} label-flag{css_disabled}">{tag_name}: {description}{status}</div>""".format(
```
Comment (dimagimon) [inline corehq/apps/hqwebapp/templatetags/hq_shared_tags.py]: E501 line too long (136 > 115 characters)

File: corehq/toggles.py [hunk 3]
```diff
@@ -1044,3 +1049,11 @@ def _commtrackify(domain_name, toggle_is_enabled):
...
     }
 )
+
+SHOW_DEV_TOGGLE_INFO = StaticToggle(
+    'highlight_feature_flags',
+    'Highlight / Mark Feature Flags in the UI',
+    TAG_DEV_TOOLS,
+    [NAMESPACE_USER]
+)
+
```
Comment (dimagimon) [inline corehq/toggles.py]: W391 blank line at end of file

File: corehq/apps/app_manager/templates/app_manager/v1/partials/case_list.html [hunk 4]
```diff
@@ -237,12 +246,15 @@
 </div>
 {% endif %}
 
+{{ request|toggle_tag_info:"CASE_LIST_LOOKUP" }}
 {% if request|toggle_enabled:"CASE_LIST_LOOKUP" %}
     {% include "app_manager/v1/partials/case_list_lookup.html" %}
     <div class="spacer"></div>
 {% endif %}
 
+{{ request|toggle_tag_info:"FIXTURE_CASE_SELECTION" }}
```
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/v1/partials/case_list.html]: Why is this here twice?

File: corehq/apps/app_manager/templates/app_manager/v2/partials/case_list.html [hunk 5]
```diff
@@ -130,6 +132,8 @@ <h4 class="panel-title panel-title-nolink">
                             <th>{% trans "Direction" %}</th>
                             {% if request|toggle_enabled:'SORT_CALCULATION_IN_CASE_LIST' %}
                             <th>{% trans "Sort Calculation" %}</th>
+                            {% elif request|toggle_enabmed:'SHOW_DEV_TOGGLE_INFO' %}
```
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/v2/partials/case_list.html]: typo in `toggle_enabmed`

File: corehq/apps/app_manager/templates/app_manager/v2/partials/case_list.html [hunk 6]
```diff
@@ -180,6 +184,8 @@ <h4 class="panel-title panel-title-nolink">
                             <td>
                                 <input class="form-control" type='text' data-bind='value: sortCalculation'/>
                             </td>
+                            {% elif request|toggle_enabmed:'SHOW_DEV_TOGGLE_INFO' %}
```
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/v2/partials/case_list.html]: typo in `toggle_enabmed`

File: corehq/apps/app_manager/templates/app_manager/v2/partials/case_list.html (hunk 5 above)
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/v2/partials/case_list.html]: yikes. thanks

File: corehq/apps/app_manager/templates/app_manager/v1/partials/case_list.html (hunk 4 above)
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/v1/partials/case_list.html]: typo. thanks

File: corehq/toggles.py [hunk 7]
```diff
@@ -40,7 +40,12 @@
     css_class='default',
     description="",  # I'm not sure...
 )
-ALL_TAGS = [TAG_ONE_OFF, TAG_EXPERIMENTAL, TAG_PRODUCT_PATH, TAG_PRODUCT_CORE, TAG_PREVIEW]
+TAG_DEV_TOOLS = Tag(
```
Comment (orangejenny) [inline corehq/toggles.py]: Nit, but `TAG_ONE_OFF` seems fine to use here and sufficient to warn people away from turning this on.

File: corehq/apps/app_manager/templates/app_manager/v1/partials/add_property_button.html [hunk 8]
```diff
@@ -6,19 +6,18 @@
         {% trans "Add Property" %}
     </button>
     {% if request|toggle_enabled:'GRAPH_CREATION' %}
-    <button class="btn btn-default dropdown-toggle" data-toggle="dropdown">
-        <span class="caret"></span>
-    </button>
-    <ul class="dropdown-menu">
-        <li data-bind="click: addProperty"><a>{% trans "Property" %}</a></li>
-        {% if request|toggle_enabled:'GRAPH_CREATION' %}
```
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/v1/partials/add_property_button.html]: Why was this `if` removed?

**Review by orangejenny — CHANGES_REQUESTED** [review]
Just a couple of small things to address.

File: corehq/apps/app_manager/templates/app_manager/v1/partials/add_property_button.html (hunk 8 above)
Comment (biyeun) [inline corehq/apps/app_manager/templates/app_manager/v1/partials/add_property_button.html]: it's inside an if with the same criteria

File: corehq/toggles.py [hunk 9]
```diff
@@ -1044,3 +1044,11 @@ def _commtrackify(domain_name, toggle_is_enabled):
...
     }
 )
+
+SHOW_DEV_TOGGLE_INFO = StaticToggle(
+    'highlight_feature_flags',
+    'Highlight / Mark Feature Flags in the UI',
+    TAG_ONE_OFF,
+    [NAMESPACE_USER]
+)
+
```
Comment (dimagimon) [inline corehq/toggles.py]: W391 blank line at end of file

**Comment by biyeun** [comment]
@orangejenny responded

**Review by orangejenny — APPROVED** [review]

**Review by benrudolph — COMMENTED** [review]
very cool, didn't review closely because it looks like jenny is taking a look

File: corehq/apps/hqwebapp/templatetags/hq_shared_tags.py [hunk 10]
```diff
@@ -219,6 +225,26 @@ def toggle_enabled(request, toggle_or_toggle_name):
...
     return _toggle_enabled(corehq.toggles, request, toggle_or_toggle_name)
 
+@register.filter
+def toggle_tag_info(request, toggle_or_toggle_name):
+    """Show Tag Information for feature flags / Toggles,
+    and if not enabled, show where the UI would be.
+    Useful for trying to find out if you have all the flags enabled in a
+    particular location or whether a feature on prod is part of a particular
+    flag. """
+    if not toggles.SHOW_DEV_TOGGLE_INFO.enabled_for_request(request):
```
Comment (snopoke) [inline corehq/apps/hqwebapp/templatetags/hq_shared_tags.py]: this toggle is so meta, love it

**Pull request merged**